### PR TITLE
ES7 client service should be only registered when ES server is up and not RED #7749

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4738619&bid=15816893)

File: modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
+package com.enonic.xp.elasticsearch.impl;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.ConnectException;
```
Comment (jsi) [inline modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java]: Codacy Issue found: [Unused import - java.net.ConnectException.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4738619)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4738619&bid=15816893)

File: modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java (hunk 1 above)
Comment (jsi) [inline modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java]: Codacy Issue found: [Unused import - java.net.ConnectException.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4738619)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4738619&bid=15816893)

File: modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    }
+
+    private void register()
+    {
+        if ( this.clientReg != null )
+        {
+            return;
+        }
+
+        this.clientReg = context.registerService( EsClient.class, client, new Hashtable<>() );
```
Comment (rymsha) [inline modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java]: don't pass empty Hashtable. pass null. Then legacy class is not in use and it is a few nanoseconds faster :)

File: modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+    private EsClient client;
+
+    private ServiceRegistration<EsClient> clientReg;
+
+    private BundleContext context;
+
+    private ScheduledExecutorService activateExecutorService;
+
+    private volatile boolean isRegistered = false;
```
Comment (rymsha) [inline modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java]: in code it looks like isRegistered = true only when clientReg is not null. you better synchronize register/unregister methods and change checking isRegistered to clientReg != null

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-elasticsearch/src/main/java/com/enonic/xp/elasticsearch/impl/ElasticsearchClientActivator.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4738619&bid=15816893)

**Pull request merged**